java-checks/src/test/files/checks/security/CryptographicKeySizeCheck.java

```diff
@@ -0,0 +1,158 @@
+  public KeyPairGenerator keyPairG;
+  private KeyGenerator keyG2;
+
+  public A() throws NoSuchAlgorithmException {
+    keyG2 = KeyGenerator.getInstance("Blowfish");
+    keyPairG = KeyPairGenerator.getInstance("RSA");
+  }
+
+  public void foo1() throws NoSuchAlgorithmException {
+    KeyGenerator keyGen = KeyGenerator.getInstance("Blowfish");
+    keyGen.init(64); // Noncompliant [[sc=5;ec=20]] {{Use a key length of at least 128 bits.}}
+  }
+
+  public void foo2() throws NoSuchAlgorithmException {
+    KeyGenerator keyGen = KeyGenerator.getInstance("Blowfish");
+    int x = 64;
+    keyGen.init(x); // FN requires following variable's values with SE-based engine
+  }
+
+  public void foo3() throws NoSuchAlgorithmException {
+    String algorithm = "Blowfish";
+    KeyGenerator keyGen = KeyGenerator.getInstance(algorithm);
+    keyGen.init(64); // FN requires following variable's values with SE-based engine
+    keyGen.getAlgorithm();
+  }
+
+  public void foo4() throws NoSuchAlgorithmException {
+    KeyGenerator keyGen;
+    keyGen = KeyGenerator.getInstance("Blowfish");
+    keyGen.init(64); // Noncompliant {{Use a key length of at least 128 bits.}}
+  }
+
+  public void foo5() throws NoSuchAlgorithmException {
+    keyG = KeyGenerator.getInstance("Blowfish");
+    keyG.init(64); // Noncompliant {{Use a key length of at least 128 bits.}}
+  }
+
+  public void foo6() {
+    keyG2.init(64); // FN requires following variable's values with SE-based engine
+  }
+
+  public void foo() throws NoSuchAlgorithmException {
+    KeyGenerator keyGen = KeyGenerator.getInstance("Blowfish");
+    KeyGenerator keyGen2 = createGen();
+
+    keyGen.init(64); // Noncompliant
+
+    keyGen2.init(96); // FN requires following variable's values with SE-based engine
+
+    KeyGenerator keyGen3 = KeyGenerator.getInstance("Blowfish");
+    keyGen3.init(64); // Noncompliant {{Use a key length of at least 128 bits.}}
+  }
+
+  public void foo7() throws NoSuchAlgorithmException {
+    keyG = KeyGenerator.getInstance("AES");
+    keyG.init(64); // Compliant
+  }
+
+  private KeyGenerator createGen() throws NoSuchAlgorithmException {
+    return KeyGenerator.getInstance("Blowfish");
+  }
+
+  public void foo8() {
+    Runnable task2 = () -> {
+      KeyGenerator keyGen;
+      try {
+        keyGen = KeyGenerator.getInstance("Blowfish");
+        keyGen.init(64); // Noncompliant
+      } catch (NoSuchAlgorithmException e) {
+        e.printStackTrace();
+      }
+    };
+  }
+
+  public void foo9() throws NoSuchAlgorithmException {
+    new Runnable() {
+      @Override
+      public void run() {
+        KeyGenerator keyGen = KeyGenerator.getInstance("Blowfish");
+        keyGen.init(64); // Noncompliant
+      }
+    };
+  }
+
+  public void foo10() throws NoSuchAlgorithmException {
+    KeyGenerator keyGen = KeyGenerator.getInstance("AES");
+    KeyGenerator keyGen2 = KeyGenerator.getInstance("Blowfish");
+    keyGen.init(64); // Compliant
+  }
+}
+
+class B {
```

Comment:
What's the difference between class A and class B? Can we find better names? There are a lot of test cases, probably too many, and it's really hard to follow how each one is specific.

Reply:
one thing I have seen with other rules is splitting test cases in multiple files.
Here class B uses the inner variable declared in class A. Class B could be moved in a different file, with a smaller class A (that wouldn't contain test cases, but only the inner variable)